ast/src/main/java/com/graphicsfuzz/common/tool/PrettyPrinterVisitor.java

```diff
@@ -460,23 +464,22 @@ public void visitWhileStmt(WhileStmt whileStmt) {
 
   @Override
   public void visitForStmt(ForStmt forStmt) {
-    out.append(indent() + "for(" + newLine());
-    out.append("    ");
+    insideForStatement = true;
+    out.append(indent() + "for(");
     visit(forStmt.getInit());
-    out.append("    " + indent());
+    out.append(" ");
     if (forStmt.hasCondition()) {
       visit(forStmt.getCondition());
     }
-    out.append(";" + newLine());
-    out.append("    " + indent());
+    out.append("; ");
     if (forStmt.hasIncrement()) {
       visit(forStmt.getIncrement());
     }
-    out.append(newLine());
-    out.append(indent() + ")" + newLine());
+    out.append(" )" + newLine());
```

Comment:
At this point you should set `insideForStatementHeader` to false, because we're done visiting the header and are now going to visit the body.  We do not want to keep this flag set while we process the body of the loop.

Reply:
Done.